rules/integrations/o365/credential_access_microsoft_365_potential_user_account_brute_force.toml

```diff
@@ -0,0 +1,174 @@
+
+### False Positive Analysis
+
+- IT administrators using automation tools (e.g., PowerShell) during account provisioning may trigger false positives if login attempts cluster.
+- Penetration testing or red team simulations may resemble spray activity.
+- Infrequent, low-volume login testing tools like ADFS testing scripts can exhibit similar patterns.
+
+### Response Recommendations
+
+- Initiate an internal incident ticket and inform the affected identity/IT team.
+- Temporarily disable impacted user accounts if compromise is suspected.
+- Investigate whether any login attempts succeeded after the spray window.
+- Block the offending IPs or ASN temporarily via firewall or conditional access policies.
+- Rotate passwords for all targeted accounts and audit for password reuse.
+- Enforce or verify MFA is enabled for all user accounts.
+- Consider deploying account lockout or progressive delay mechanisms if not already enabled.
+"""
+references = [
+    "https://learn.microsoft.com/en-us/security/operations/incident-response-playbook-password-spray",
+    "https://learn.microsoft.com/en-us/purview/audit-log-detailed-properties",
+    "https://securityscorecard.com/research/massive-botnet-targets-m365-with-stealthy-password-spraying-attacks/",
+    "https://github.com/0xZDH/Omnispray",
+    "https://github.com/0xZDH/o365spray",
+]
+risk_score = 47
+rule_id = "26f68dba-ce29-497b-8e13-b4fde1db5a2d"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: SaaS",
+    "Data Source: Microsoft 365",
+    "Data Source: Microsoft 365 Audit Logs",
+    "Use Case: Identity and Access Audit",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+FROM logs-o365.audit-*
+
+| MV_EXPAND event.category
+| EVAL
+    time_window = DATE_TRUNC(5 minutes, @timestamp),
+    user_id = TO_LOWER(o365.audit.UserId),
+    ip = source.ip,
+    login_error = o365.audit.LogonError,
+    request_type = TO_LOWER(o365.audit.ExtendedProperties.RequestType),
+    asn_org = source.`as`.organization.name,
+    country = source.geo.country_name,
+    event_time = @timestamp
+
+| WHERE event.dataset == "o365.audit"
+  AND event.category == "authentication"
+  AND event.provider IN ("AzureActiveDirectory", "Exchange")
+  AND event.action IN ("UserLoginFailed", "PasswordLogonInitialAuthUsingPassword")
+  AND request_type RLIKE "(oauth.*||.*login.*)"
+  AND login_error != "IdsLocked"
+  AND login_error NOT IN (
+    "EntitlementGrantsNotFound", "UserStrongAuthEnrollmentRequired", "UserStrongAuthClientAuthNRequired",
+    "InvalidReplyTo", "SsoArtifactExpiredDueToConditionalAccess", "PasswordResetRegistrationRequiredInterrupt",
+    "SsoUserAccountNotFoundInResourceTenant", "UserStrongAuthExpired", "CmsiInterrupt"
+  )
+  AND user_id != "not available"
+  AND o365.audit.Target.Type IN ("0", "2", "6", "10")
+  AND asn_org != "MICROSOFT-CORP-MSN-AS-BLOCK"
```

Comment:
what if they bruteforce from a VM in azure ?

Reply:
yea, that is a very valid concern. Wouldn't take much to load up a VM from their own tenant and do this. I'll remove for now then as I think some of the other conditions should be fine for now.